mad-dog/src/tests/test_plays.js

```diff
@@ -0,0 +1,155 @@
+    // use nonce to bypass cache
+    url: `http://localhost:5000/v1/metrics/plays?bucket_size=century&nonce=${Math.random()}`
+  })).data.data[0] || {}).count || 0
+}
+
+async function submitTrackListen (executeOne, trackId, userId, solanaListen) {
+  const initialPlays = getTotalPlays()
+
+  let signature
+  try {
+    await executeOne(0, async (libs) => {
+      const start = Date.now()
+      const identityResponse = await logTrackListen(trackId, userId, solanaListen)
+      signature = identityResponse.solTxSignature
+      console.log(`Logged track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen}) | Processed in ${Date.now() - start}ms`)
+    })
+  } catch (err) {
+    console.log(`Failed to log track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen}) with error ${err}`)
+    return false
+  }
+
+  const pollStart = Date.now()
+  console.log(`Polling track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen})`)
+
+  if (solanaListen) {
+    let resp = (await getSolPlay(signature)).data
+    while (!resp.data) {
+      await delay(500)
+      resp = (await getSolPlay(signature)).data
+      if (Date.now() - pollStart > MaxPollDurationMs) {
+        throw new Error(`Failed to find ${signature} for userId=${userId}, trackId=${trackId} in ${MaxPollDurationMs}ms`)
+      }
+    }
+  } else {
+    let plays = await getTotalPlays()
+    while (plays === initialPlays) {
+      await delay(500)
+      plays = await getTotalPlays()
+      if (Date.now() - pollStart > MaxPollDurationMs) {
+        throw new Error(`Failed to find listen for userId=${userId}, trackId=${trackId} in ${MaxPollDurationMs}ms`)
+      }
+    }
+  }
+
+  console.log(`Found track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen}) in discovery-provider`)
+
+  return true
+}
+
+async function trackListenCountsTest ({ executeOne }) {
+  const numBaseTrackListens = 10
+  const numAnonBaseTrackListens = 10
+  const numSolanaTrackListens = 10
+  const numAnonSolanaTrackListens = 10
+
+  const start = Date.now()
+
+  const numSuccessfulSolanaTrackListens = (await Promise.all(
+    Array.from({ length: numSolanaTrackListens }, async () => {
+      const numListens = Math.floor(Math.random() * 5) + 1
+      const trackId = Math.floor(Math.random() * 10000000)
+      const userId = Math.floor(Math.random() * 10000000)
+
+      return (await Promise.all(
+        Array.from({ length: numListens }, () =>
+          submitTrackListen(executeOne, trackId, userId, true)
+        )
+      )).reduce((a, b) => a + b, 0)
+    })
+  )).reduce((a, b) => a + b, 0)
+
+  const numSuccessfulAnonSolanaTrackListens = (await Promise.all(
+    Array.from({ length: numAnonSolanaTrackListens }, async () => {
+      const numListens = Math.floor(Math.random() * 5) + 1
+      const trackId = Math.floor(Math.random() * 10000000)
+      const userId = `anon-${Math.floor(Math.random() * 10000000)}`
```

Comment:
this is actually incorrect - we need to send _no_ userId and confirm the play is still written

Reply:
actually nvm, this is okay - better to use a uuid but this serves the purpose https://github.com/AudiusProject/audius-protocol/blob/48f8eb4d16db8a622c4b84973f3dc469401d729b/libs/src/api/track.js#L530